### PR TITLE
Adding option --connection for db commands

### PR DESCRIPTION
**Pull request by jalogut**
If set, this option allows to select a specific connection database on projects with several databases

### CODE REVIEW

File: src/N98/Magento/Command/Database/AbstractDatabaseCommand.php [hunk 1]
```diff
@@ -113,6 +152,9 @@ protected function resolveTables(array $excludes, array $definitions, array $res
      */
     protected function getDatabaseHelper()
     {
-        return $this->getHelper('database');
+        if (!$this->dbHelper) {
```
Comment (ktomk) [inline src/N98/Magento/Command/Database/AbstractDatabaseCommand.php]: No need for that code, getHelper stores all helpers as singleton.

File: src/N98/Util/Console/Helper/DatabaseHelper.php [hunk 2]
```diff
@@ -32,6 +32,21 @@ class DatabaseHelper extends AbstractHelper
     protected $_tables;
 
     /**
+     * @var string
+     */
+    protected $_connectionType = 'default';
```
Comment (ktomk) [inline src/N98/Util/Console/Helper/DatabaseHelper.php]: Please do not signal visibility with the underscore character (even we have code that does it wrong in the project).

File: src/N98/Util/Console/Helper/DatabaseHelper.php [hunk 3]
```diff
@@ -51,11 +66,11 @@ public function detectDbSettings(OutputInterface $output)
             return;
         }
 
-        if (!isset($config['db']['connection']['default'])) {
-            throw new RuntimeException('Cannot find default connection config in app/etc/config.php');
+        if (!isset($config['db']['connection'][$this->_connectionType])) {
+            throw new RuntimeException(sprintf('Cannot find "%s" connection config in app/etc/config.php', $this->_connectionType));
```
Comment (ktomk) [inline src/N98/Util/Console/Helper/DatabaseHelper.php]: I think this is now env.php with Magerun2 (existing mistake), we should fix it along the lines.

File: src/N98/Util/Console/Helper/DatabaseHelper.php (hunk 2 above)
Comment (ktomk) [inline src/N98/Util/Console/Helper/DatabaseHelper.php]: This works fine as  a private member.

File: src/N98/Magento/Command/Database/AbstractDatabaseCommand.php [hunk 4]
```diff
@@ -5,11 +5,23 @@
...
+use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputOption;
 
 abstract class AbstractDatabaseCommand extends AbstractMagentoCommand
 {
     /**
+     * @var string
+     */
+    protected $connectionTypeOption = 'connection';
```
Comment (ktomk) [inline src/N98/Magento/Command/Database/AbstractDatabaseCommand.php]: I think we can spare this member (or make it private).

File: src/N98/Magento/Command/Database/AbstractDatabaseCommand.php [hunk 5]
```diff
@@ -5,11 +5,23 @@
...
+
+    /**
+     * @var
+     */
+    protected $dbHelper;
```
Comment (ktomk) [inline src/N98/Magento/Command/Database/AbstractDatabaseCommand.php]: Not needed (see below comment on getDatabaseHelper() method)

**Comment by ktomk** [comment]
Thanks for your PR! I've added some comments while reviewing. Let me know if you've got any questions about that or you want more information for a particular change.

**Comment by jalogut** [comment]
@ktomk thanks for the feedback. I think that I changed all points requested. Hope it's better now.

**Comment by tkn98** [comment]
thanks for the quick update! Can you squash the commit together with your previous one and force-push? That should keep the changes in one commit. If you don't know what I'm talking about, just let me know, then I'll take care of it.

**Comment by jalogut** [comment]
Done. Nice git feature to squash commits, I did not know that. Thanks for the feedback.

**Comment by tkn98** [comment]
While I keep you busy ;-P, have you considered to take a look why the actual build failed? https://travis-ci.org/netz98/n98-magerun2/jobs/174146650#L547 <- This should be trivial to fix, but I'm most of  interested if you have seen this before or not. It's relatively new with the build tools we use, that's why I'm curious in your unfiltered feedback.

**Comment by jalogut** [comment]
Hi, I did not notice the build failed. I did not get any email notification about it. I am checking the issue now, but I do not really know what the problem is. Do you know why it fails?

**Comment by jalogut** [comment]
I just updated the PR with the right order for the "use" classes. Hope that was the only issue

**Comment by tkn98** [comment]
We believe that the feature requested is available in the development version. It can be upgraded to it using the `self-update` command with the `--unstable` switch.
---
Thanks! Please give us a little time to update the changelog file.

**Comment by jalogut** [comment]
Thanks for merging :)

**Comment by tkn98** [comment]
While you changed all db commands, not all of the commands you changed do reflect the changes properly, for example, the db:info command has the parameter but not in use. Just FYI so far.

**Comment by jalogut** [comment]
@tkn98 Ok, no problem.

**Pull request merged**